src/azfswatchdog

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash 
+
+#  
```

Comment:
We have to set some ground rules.
1. The shared data between the mount helper script and the azfs watchdog service will reside in a well known directory inside the tmpfs /run/. We should make a directory named /run/azblobnfs/ and inside that have a file named say, /run/azblobnfs/mountmap.
2. The responsibility of creatin the above directory and the mountmap file lies with the azfs watchdog service, so if the directory and mountmap file doesn't exist the mount helper may bail out with an appropriate error asking the user to check.
3. Since this file would be updated by the service and the mount helper, they should update the file after taking exclusive lock (use the bash command flock for that). Write a function like the following

update_mountmap()
{
    flock /run/azblobnfs/mountmap -c "eval $*";
}

now callers can call it as

update_mountmap sed -i "/ $l_ip $l_nfsip/d" $MOUNTMAP

which will safely update the file
5. chattr will not work for tmpfs so we will have to get rid of that.
6. Once the initial code takes shape, we can think of ways to protect this file from external/undesired mods.
7. All the logs from these files must also go to a log file so that any issues can be debugged. We can start with a log file named /run/azblobnfs/log. In later iterations we will have to think of ways to better rotate this logfile but for now let's keep it simple and start with this single file. We need to write functions for logging, which we can then use in the script instead of the plain echo.
8. f.e. check out the pecho/secho/becho/sbecho/etc variants in 
https://msazure.visualstudio.com/One/_git/Storage-XDataMove-Agent-Linux?path=/packaging/vmrootfs/usr/bin/xdmsh&_a=contents&version=GBmain

Reply:
updated handling of mountmap in this iteration. Will add proper logging in next one.

Reply:
Added code to send all the logs into one file which will be in /run/azblobnfs/log as you suggested. To add the shared code between the files in src, I have added "lib" folder in top level which will contain all the helping functions common among the scripts in "src" folder.